File: src/main/java/com/xgarage/app/dto/SupplierDto.java
package com.xgarage.app.dto;

import com.xgarage.app.model.Location;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SupplierDto {
    private Long Id;
    private Long userId;
    private String name;
    private String email;
    private String cr;
    private String contactName;
    private String phone;
    Set<Location> locations;
    private boolean enabled;
    private long submittedBids = 0;
    private long completedDeals = 0;
    private double rating;
}

File: src/app/xgarage/core/dto/supplierdto.ts
export interface SupplierDto{
    id?: number;
    userId?: number;
    name?: string;
    cr?: string;
    contantName?: string;
    phone?: string;
    locations?: Location[];
    enabled?: boolean;
    submittedBids?: number;
    completedDeals?: number;
    rating?: number;
}